src/main/java/org/littleshoot/proxy/impl/ClientToProxyConnection.java

```diff
@@ -770,6 +771,15 @@ protected void exceptionCaught(Throwable cause) {
         }
     }
 
+    @Override
+    public void channelRegistered(ChannelHandlerContext ctx) throws Exception {
+        try {
+            this.proxyServer.getGlobalStateHandler().persistToChannel(ctx.channel());
+        } finally {
+            super.channelRegistered(ctx);
+        }
+    }
+
```

Comment:
This is a client to proxy connection. A moment where a client connects to the proxy and a channel is registered. We can generate a trace id here and put it to channel attributes so we can restore it into MDC later.